src/kaggle/api/kaggle_api_extended.py

```diff
@@ -374,6 +469,28 @@ def __repr__(self):
 
 
 class KaggleApi:
+    """
+    KaggleApi provides methods for interacting with Kaggle's public API.
+
+    This class manages authentication, configuration, and communication with Kaggle endpoints
+    for datasets, competitions, kernels, models, and more. It supports downloading and uploading
+    datasets, managing competition submissions, handling kernels (notebooks and scripts), and
+    querying Kaggle resources.
+
+    Configuration is handled via environment variables or a configuration file, and the class
+    supports both API key and OAuth authentication methods. It validates input parameters for
+    various Kaggle resource types and manages local paths and proxy settings.
+
+    Usage:<br>
+        api = KaggleApi()<br>
+        api.authenticate()<br>
+        api.dataset_download_files('username/dataset-name')<br>
+        api.competition_submit('submission.csv', 'My submission', 'competition-name')<br>
```

Comment:
Does there need to be html here?

Reply:
Without the `br`s the "Usage" section was being generated as a single paragraph. But that was before switching to sphinx. I'll check if they are still needed; thanks for the call-out.